model.py

```diff
@@ -74,81 +79,85 @@ def __init__(self, *args, envelope=poly, IP_hs=50, icen=0):
         # IP_hs: Half size of the IP (number of sampling knots).
         # icen : Central pixel (to center polynomial for numeric reason).
         self.icen = icen
-        self.S_star, self.uj, self.iod_j, self.atmj, self.IP = args
+        self.S_star, self.lnwave_cell, self.lnspec_cell, self.spec_atm, self.IP = args
         # convolving with IP will reduce the valid wavelength range
-        self.dx = self.uj[1] - self.uj[0]  # sampling in uniform resampled Iod
+        self.dx = self.lnwave_cell[1] - self.lnwave_cell[0]  # sampling in uniform resampled Iod
         self.IP_hs = IP_hs
         self.vk = np.arange(-IP_hs, IP_hs+1) * self.dx * c
-        self.uj_eff = self.uj[IP_hs:-IP_hs]
+        self.lnwave_cell_eff = self.lnwave_cell[IP_hs:-IP_hs]
         self.envelope = envelope
         #print("sampling [km/s]:", self.dx*c)
 
-    def __call__(self, i, v, a, b, s,t, cc=[0]):
+    def __call__(self, pixel, par_rv, par_norm, par_wave, par_ip, par_atm, par_bkg=[0]):
         # wavelength solution 
         #    lam(x) = b0 + b1 * x + b2 * x^2
-        ui = np.log(np.poly1d(b[::-1])(i-self.icen))
+        lnwave_obs = np.log(np.poly1d(par_wave[::-1])(pixel-self.icen))
 
         # IP convolution
-        if len(self.atmj) == 0:
-            Sj_eff = np.convolve(self.IP(self.vk, *s), self.S_star(self.uj-v/c) * (self.iod_j + cc[0]), mode='valid')
+        if len(self.spec_atm) == 0:
+            Sj_eff = np.convolve(self.IP(self.vk, *par_ip), self.S_star(self.lnwave_cell-par_rv/c) * (self.lnspec_cell + par_bkg[0]), mode='valid')
         else:
             # telluric forward modelling
-            atm = np.ones(len(self.atmj[0]))
-            for aj in range(0,len(self.atmj),1):
-                atm *= (self.atmj[aj]**np.abs(t[aj]))
+            atm = np.ones(len(self.spec_atm[0]))
+            for aj in range(0,len(self.spec_atm),1):
+                atm *= (self.spec_atm[aj]**np.abs(par_atm[aj]))
 
 	    # variable telluric wavelength shift; one shift for all molecules
-            if len(t) == len(self.atmj)+1:
-                atm = np.interp(self.uj, self.uj-np.log(1+t[-1]/c), atm)
+            if len(par_atm) == len(self.spec_atm)+1:
```

Comment:
I am also not totally happy with the current solution, but wanted to avoid to come along with too much new variables. Maybe dict is an option. Have to think about it.